src/tests/JIT/SIMD/ShiftOperations.cs

```diff
@@ -0,0 +1,501 @@
+{
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static uint Shlx32bit(uint x, int y) => x<< y;
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static int Sarx32bit(int x, int y) => x >> y;
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static uint Shrx32bit(uint x, int y) => x >> y;
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static uint Ror(uint x) => BitOperations.RotateRight(x, 2);
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static ulong Shlx64bit(ulong x, int y) => x << y;
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static long Sarx64bit(long x, int y) => x >> y;
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static ulong Shrx64bit(ulong x, int y) => x >> y;
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static unsafe ulong ShrxRef64bit(ulong *x, int y) => *x >> y;
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static ulong Rorx(ulong x) => BitOperations.RotateRight(x, 2);
+
+    public static unsafe int Main()
+    {
+        int returnCode = 100;
+
+        try
+        {
+            uint  valUInt = 0;
+            int   valInt = 0;
+            ulong valULong = 0;
+            long valLong = 0;
+            int shiftBy = 0;
+            uint resUInt = 0;
+            int resInt = 0;
+            ulong resULong = 0;
+            long resLong = 0;
+            uint expectedUInt = 0;
+            int expectedInt = 0;
+            ulong expectedULong = 0;
+            long expectedLong = 0;
+            int MOD32 = 32;
+            int MOD64 = 64;
+
+            //
+            // Shlx32bit tests
+            //
+            valUInt = 0;
+            shiftBy = 1;
+            resUInt = Shlx32bit(valUInt, shiftBy);
+            expectedUInt = (uint) (valUInt * Math.Pow(2, (shiftBy % MOD32)));
+            Console.Write("UnitTest Shlx32bit({0},{1}): {2}", valUInt, shiftBy, resUInt);
+            if (resUInt != expectedUInt)
+            {
+                Console.Write(" != {0} Failed.\n", expectedUInt);
+                returnCode = 101;
+            }
+            Console.Write(" == {0} Passed.\n", expectedUInt);
+
+            valUInt = 8;
+            shiftBy = 1;
+            resUInt = Shlx32bit(valUInt, shiftBy);
+            expectedUInt = (uint) (valUInt * Math.Pow(2, (shiftBy % MOD32)));
+            Console.Write("UnitTest Shlx32bit({0},{1}): {2}", valUInt, shiftBy, resUInt);
+            if (resUInt != expectedUInt)
+            {
+                Console.Write(" != {0} Failed.\n", expectedUInt);
+                returnCode = 101;
+            }
+            Console.Write(" == {0} Passed.\n", expectedUInt);
+
+            valUInt = 1;
+            shiftBy = 31;
+            resUInt = Shlx32bit(valUInt, shiftBy);
+            expectedUInt = (uint) (valUInt * Math.Pow(2, (shiftBy % MOD32)));
+            Console.Write("UnitTest Shlx32bit({0},{1}): {2}", valUInt, shiftBy, resUInt);
```

Comment:
Why not create a `Validate()` function that will do most of these things at one place?

```c#
public int Validate(..., actual, ...) {
  expected = 
  if (expected != actual) { 
    Console.WriteLine("Fail");
    return 101;
  }
  return 100;
}
```

Reply:
You can then just have an input array and iterate over it or something like that.